### PR TITLE
Refactor tsp-client to use yargs

### PR DESCRIPTION
**Pull request by tjprescott**
This PR converts tsp-client to use `yargs` instead of `node:util` for parsing, as it is a full-fledged CLI tool and benefits from the yargs framework.

### CODE REVIEW

File: tools/tsp-client/src/index.ts [hunk 1]
```diff
@@ -243,111 +382,236 @@ async function convert(readme: string, outputDir: string, arm?: boolean): Promis
...
+      process.exit(1);
+    }
+    outputDir = resolvePath(normalizePath(newOutputDir));
   }
+  Logger.info(`Using output directory '${outputDir}'`);
+  return outputDir;
 }
 
-main().catch((err) => {
+const parser = yargs(hideBin(process.argv))
```
Comment (catalinaperalta) [inline tools/tsp-client/src/index.ts:457]: I'm not seeing the --version flag anymore, was it removed?

**Review by catalinaperalta — COMMENTED** [review]
This looks good :) I think we should wait to merge this until I have the test script checked in to make sure none of the main usage scenarios are broken since this is a pretty big change. Also, since we're changing the generate lock file functionality from a flag to a command we should bump tsp-client version to 0.10.0

File: tools/tsp-client/src/log.ts [hunk 2]
```diff
@@ -54,46 +56,7 @@ export function printBanner() {
...
-                            generated files. Default is your current directory  [string]
-  --repo                    Repository where the project is defined for init 
-                            or update                                           [string]
-  -v, --version             Show version number                                 [boolean]
-`;
-export function printUsage() {
-  Logger(usageText);
-}
-
-export async function printVersion() {
```
[the remark this answers is not in the document]
Comment (tjprescott) [reply tools/tsp-client/src/log.ts:96]: Yes, it does.

File: tools/tsp-client/src/index.ts [hunk 3]
```diff
@@ -1,45 +1,75 @@
...
+}
+
+async function initCommand(argv: any) {
+  let outputDir = resolveOutputDir(argv);
+  let tspConfig = argv["tsp-config"];
+  const skipSyncAndGenerate = argv["skip-sync-and-generate"];
+  const commit = argv["commit"];
+  const repo = argv["repo"];
+
+  commandPreamble(argv);
```
[the remark this answers is not in the document]
Comment (tjprescott) [reply tools/tsp-client/src/index.ts:40]: We could, it was just replacing the lines in the exiting code that were printing the header/version/etc. I don't think you'd want to print the preamble if you ran `tsp-client --help` though.  Whatever you want it to do :)

File: tools/tsp-client/src/index.ts (hunk 1 above)
Comment (tjprescott) [reply tools/tsp-client/src/index.ts:457]: It is automatically supported by the framework. 
image

File: tools/tsp-client/src/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (tjprescott) [reply tools/tsp-client/src/index.ts:40]: It _did_ have main, but it became so small I just took it out.

[the remark this answers is not in the document]
Comment (catalinaperalta) [reply tools/tsp-client/src/index.ts:40]: Ah I see, ok well it's alright for now where it is.

**Review by catalinaperalta — APPROVED** [review]

**Pull request merged**